.github/workflows/main.yml

```diff
@@ -1,6 +1,24 @@
 name: tests
 
-on: push
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types:
+      - assigned
+      - unassigned
+      - labeled
+      - unlabeled
+      - opened
+      - edited
+      - reopened
+      - synchronize
+      - ready_for_review
+      - locked
+      - unlocked
+      - review_requested
+      - review_request_
```

Comment:
Before the tests and linting used to run for any push from any branch

Now they run for any push in the master branch and for any change in a PR except when the PR is a draft (Thanks to this [line](https://github.com/portugueslab/stytra/blob/d1b38cbba60313e3f2ed0629b5bdcd44de7f77ad/.github/workflows/main.yml#L27))

@vigji & @vilim What do you think about it?
I couldn't find a solution to avoid running the workflows on every push avoiding only the case when the PR is in draft mode, and I'm not sure that would be even desired (that = running in every push from any branch)

Reply:
Interesting that doesn't seem to be possible. What you proposed looks fine to me